### PR TITLE
[FIX] website_sale_options: display the options name with the linked …

### PR DESCRIPTION
**Pull request by Gorash**
…product in the description after quantity changed

### CODE REVIEW

File: addons/website_sale_options/models/sale_order.py [hunk 1]
```diff
@@ -33,26 +33,30 @@ def _cart_find_product_line(self, cr, uid, ids, product_id=None, line_id=None, c
...
-
-            if linked_line_id and linked_line_id in map(int,so.order_line):
-                linked = sol.browse(cr, SUPERUSER_ID, linked_line_id, context=context)
-                line.write({
-                        "name": _("%s\nOption for: %s") % (line.name, linked.product_id.name_get()[0][1]),
-                        "linked_line_id": linked_line_id
-                    })
+        # link a product to the sale order
+        if kwargs.get('linked_line_id'):
+            linked_line_id = sol.browse(cr, SUPERUSER_ID, kwargs.get('linked_line_id'), context=context)
```
Comment (beledouxdenis) [inline addons/website_sale_options/models/sale_order.py]: Just a note: As you ensured just above `kwargs.get('linked_line_id')`, you can safely use `kwargs['linked_line_id']`. This doesn't change much, but it is kind of an Odoo guildeline. This is prettier as well as shorter.

File: addons/website_sale_options/models/sale_order.py [hunk 2]
```diff
@@ -33,26 +33,30 @@ def _cart_find_product_line(self, cr, uid, ids, product_id=None, line_id=None, c
...
+            line.write({
+                    "name": _("%s\nOption for: %s") % (line.name, linked_line_id.product_id.name_get()[0][1]),
+                    "linked_line_id": linked_line_id.id
+                })
 
+        value['option_ids'] = set()
+        for so in self.browse(cr, uid, ids, context=context):
             # select linked product
-            option_ids = [l for l in so.order_line if l.linked_line_id.id == line.id]
+            option_line_ids = [l for l in so.order_line if l.linked_line_id.id == line.id]
```
Comment (beledouxdenis) [inline addons/website_sale_options/models/sale_order.py]: I took 10 minutes do understand that one. While it seems fine, you should probably add a short comment to say what `option_line_ids` is.

File: addons/website_sale_options/models/sale_order.py [hunk 3]
```diff
@@ -33,26 +33,30 @@ def _cart_find_product_line(self, cr, uid, ids, product_id=None, line_id=None, c
...
 
             # update line
-            for l in option_ids:
-                super(sale_order, self)._cart_update(cr, uid, ids, l.product_id.id, l.id, add_qty, set_qty, context=context, **kwargs)
+            for option_line_id in option_line_ids:
+                super(sale_order, self)._cart_update(cr, uid, ids, option_line_id.product_id.id, option_line_id.id, add_qty, set_qty, context=context, **kwargs)
+                option_line_id.write({"name": _("%s\nOption for: %s") % (option_line_id.name, option_line_id.linked_line_id.product_id.name_get()[0][1])})
+
+            value['option_ids'] |= set(map(int, option_line_ids))
```
Comment (beledouxdenis) [inline addons/website_sale_options/models/sale_order.py]: This one is just unbearable. can't you just do one of the two below:
```
value['option_ids'].union(option_line_ids.ids)
``` 
or, within the for loop just above
```
value['option_ids'].add(option_line_id.id)
``` 
Your code must always be the most readable possible for a newbie. Always thing that someone will have to read your code after you wrote it.

**Comment by beledouxdenis** [comment]
I made a few comments. That's mainly coding style, the PR seems fine otherwise :+1:.

Maybe add in the commit message what was not working. 

In general, when you fix something, do not say what you do in the revision, the code should be enough clear to cover that, instead say what was not working, with an example if the description alone is not enough.

**Comment by Gorash** [comment]
Hi,
I made the changes
Thanks

**Comment by beledouxdenis** [comment]
Seems right :). Thanks !

**Pull request merged**